### PR TITLE
Add extractors for SiNAPS Research Platform

### PR DESCRIPTION
**Pull request by NinelK**
Contains extractors from binary format (raw + filtered + aux channels) as well as an additional extractor from *.h5 format (only filtered channels)

### CODE REVIEW

**Comment by alejoe91** [comment]
@NinelK can you allow edits from maintainers? The pre-commit bot failed to refactor the code according to black

**Comment by NinelK** [comment]
> @NinelK can you allow edits from maintainers? The pre-commit bot failed to refactor the code according to black

it was allowed in the settings on the side, not sure why pre-comit.ci can not push.
I ran black locally, and it seems like this resolves the issue.

File: src/spikeinterface/extractors/sinapsrecordingh5extractor.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import numpy as np
+
+from probeinterface import get_probe
+
+from ..core.core_tools import define_function_from_class
+from ..core import BaseRecording, BaseRecordingSegment
+from ..preprocessing import UnsignedToSignedRecording
+
+
+class SinapsResearchPlatformH5RecordingExtractor_Unsigned(BaseRecording):
```
Comment (alejoe91) [inline src/spikeinterface/extractors/sinapsrecordingh5extractor.py:11]: Maybe we can just apply the unsigned_to_signed in the get_traces, which is simpler to read and will get rid of the complex class interplay. I can implement that

Comment (alejoe91) [reply src/spikeinterface/extractors/sinapsrecordingh5extractor.py]: done

**Comment by alejoe91** [comment]
@NinelK after discussing with @samuelgarcia, we decided that in this case it's worth adding the unsigned to signed scaling directly to the SinapsH5Extractors (avoiding that class trick, which makes it less readable).

Also moved everything to a single file and cleaned docs and helper functions. Ready to merge on my end

**Review by alejoe91 — APPROVED** [review]

**Comment by alejoe91** [comment]
@NinelK can you test it one more time on your side? I don't have h5 test data

**Comment by NinelK** [comment]
All works now, just fixed 1 issue with a missing argument

**Pull request merged**